init.el

```diff
@@ -58,22 +58,21 @@ straight.el or Guix depending on the value of
 (mkdir rational-config-etc-directory t)
 (mkdir rational-config-var-directory t)
 
-;; The custom file
-(customize-set-variable 'custom-file
-                        (expand-file-name "custom.el" rational-config-path))
-
-(defun rational-load-custom-file ()
-  "Load the `custom-file' after saving customized values.
-
-Customized values set during the initialization process are
-saved, other values set through the Customization UI or via
-certain workflows are not lost in this process. The `custom-file'
-is loaded last."
-
-  (customize-save-customized)
-  (load custom-file t))
-
-(add-hook 'after-init-hook #'rational-load-custom-file)
+;; Here the `custom.el' file is used by the Customization UI to store
+;; value-setting forms in a customization file, rather than at the end
+;; of the `init.el' file. The file is loaded after this `init.el'
+;; file, after the user `config.el' has been loaded. This means that
+;; any set variables in the user `config.el' will be overridden with
+;; the values set with the Customization UI. Rather than expanding the
+;; `custom-file' variable when adding the hook, the return value of
+;; the form that sets `custom.el' is used.
+(add-hook 'after-init-hook
+          (lambda ()
```

Comment:
The problem here is the following: 1) its a bit challenging to remove this hook if the user does not want to load the `custom-file` at all and 2) if they do manage to get the hook removed, then the `custom-file` value is never set so, customizations end up in `init.el`. We'd prefer to move those to the custom.el file so there are no future merge conflicts in init.el at least.

Reply:
I think a *rational* way to improve this would be to avoid a hook, and instead use a new customization variable in `early-init.el`. At the moment, my changes to the README mention customizations made in `early-init.el`, but in fact there are none (I made an assumption).

We can opt for the following:
1. Prediate loading of the custom file on the boolean value of `rational-load-custom-file`
2. Barf the load form out of the hook and wrap it in a when form
3. Move the new form to the bottom of `init.el`

This preserves the current algorithm:
1. Load `early-init.el`
2. Load `early-config.el` if it exists
3. Load `init.el`, and before we reach the end of the file proceed to step four
4. Load `config.el`
5. Load `custom.el` if it exists (and with this change, if the user *wants* to load a custom file)
6. Set GC cons threshold and finish initializtion

I'll make a new commit, because I think avoiding the use of a hook will permit more flexibility in `config.el`; users can then opt-in to loading a custom file, and having any customizations in `config.el` overwritten (unless they're aware of that and want this, or have no conflicting customizations).

Opt-in would be the best approach, I think, and with that we don't need the hook because it'll happen according to the algorithm above regardless, so we don't need `after-init-hook` even now.